platform/qt/src/qmapboxgl.cpp

```diff
@@ -1005,23 +1225,55 @@ void QMapboxGL::addLayer(const QVariantMap &params)
     d_ptr->mapObj->addLayer(std::move(*layer));
 }
 
+/*!
+    Removes the layer \a id.
+*/
 void QMapboxGL::removeLayer(const QString& id)
 {
     d_ptr->mapObj->removeLayer(id.toStdString());
 }
 
-void QMapboxGL::addImage(const QString &name, const QImage &sprite)
+/*!
+    Adds the \a image with the identifier \a id that can be used
+    later by a symbol layer.
+
+    \sa addLayer()
+*/
+void QMapboxGL::addImage(const QString &id, const QImage &image)
 {
-    if (sprite.isNull()) return;
+    if (image.isNull()) return;
 
-    d_ptr->mapObj->addImage(name.toStdString(), toSpriteImage(sprite));
+    d_ptr->mapObj->addImage(id.toStdString(), toSpriteImage(image));
 }
 
-void QMapboxGL::removeImage(const QString &name)
+/*!
+    Removes the image \a id.
+*/
+void QMapboxGL::removeImage(const QString &id)
 {
-    d_ptr->mapObj->removeImage(name.toStdString());
+    d_ptr->mapObj->removeImage(id.toStdString());
 }
 
+/*!
+    Adds a \a filter_ to a style \a layer_ accordingly to the \l
+    {https://www.mapbox.com/mapbox-gl-style-spec/#types-filter}{the vector tiles specification}.
+
+    Given a layer \b "marker" from an arbitrary GeoJSON source containing features of type \b
```

Comment:
There are two instances each of “GeoJSON source” and “GeoJSON data source”. Best to be consistent, so that developers don’t think there may be a distinction between the two.